public/docs/_examples/dependency-injection/dart/lib/test_component.dart

```diff
@@ -0,0 +1,62 @@
+// Simulate a simple test
+
+// Reader should look to the testing chapter for the real thing
+import 'package:angular2/core.dart';
+import 'heroes/hero_service.dart';
+import 'heroes/hero_list_component.dart';
+
+@Component(
+    selector: 'my-tests',
+    template: '''
+    <h2>Tests</h2>
+    <p id="tests">Tests {{results['pass']}}: {{results['message']}}</p>
+  ''')
+class TestComponent {
+  var results = runTests();
+}
+
+/////////////////////////////////////
+//#docregion spec
+var expectedHeroes = [
+  {'name': 'A'},
+  {'name': 'B'}
+];
+
+class HeroServiceMock implements HeroService {
+  @override
+  getHeroes() => expectedHeroes;
+}
+
+runTests() {
+  var mockService = new HeroServiceMock();
+  it('should have heroes when HeroListComponent created', () {
+    var hlc = new HeroListComponent(mockService);
+    expect(hlc.heroes.length)['toEqual'](expectedHeroes.length);
+  });
+  return testResults;
+}
+//#enddocregion spec
+
+//////////////////////////////////
+
+// Fake Jasmine infrastructure
+String testName;
```

Comment:
Quite easy

just a few bits about how it looks in Dart. If more info is needed just ping me
```dart
import 'package:test/test.dart';
import 'package:angular2/angular2.dart';
import 'package:angular2_testing/angular2_testing.dart';

void main() {
  initAngularTests();
  group('basic', () {
    ngTest('single selection', (TestComponentBuilder tcb) async {
      ComponentFixture tc = await tcb.createAsync(SingleConfigured);
      tc..detectChanges();
      BwuArraySelector el =
          (tc.componentInstance as SingleConfigured).arraySelector;
      // Nothing selected
      expect(el.selectedItem, null);
      expect(el.isSelected(el.items[0]), isFalse);
      expect(el.isSelected(el.items[1]), isFalse);
```